crates/lib-dialects/src/tsql.rs

```diff
@@ -83,6 +91,18 @@ pub fn raw_dialect() -> Dialect {
         "equals",
     );
 
+    // T-SQL specific lexer patches:
+    // 1. T-SQL only uses -- for inline comments, not # (which is used in temp table names)
+    // 2. Update word pattern to allow # at the end (SQL Server 2017+ syntax)
```

Comment:
I am just able to test this with SQL Server 2017, it is the oldest I have access to.
I am sure it also worked with 2008 too.